cvereporter/fetch_vulnerabilities.py

```diff
@@ -0,0 +1,85 @@
+from datetime import datetime
+from cyclonedx.model.impact_analysis import ImpactAnalysisAffectedStatus
+from cyclonedx.model.vulnerability import Vulnerability, VulnerabilitySource,VulnerabilityScoreSource, VulnerabilityRating, VulnerabilitySeverity, BomTarget, BomTargetVersionRange
+
+
+parser = argparse.ArgumentParser(description='Scrape the OpenJDK CVE site and return it in JSON format.')
+
+parser.add_argument('--date', type=str, default='2023-01-17', help='The date to scrape CVEs for (YYYY-MM-DD)')
+
+
+args = parser.parse_args()
+def fetch_cves(date: str)->list[Vulnerability]:
+    vulnerabilities = []
+    # fetch the CVEs for the given date
+    url = 'https://openjdk.org/groups/vulnerability/advisories/' + args.date
+    r = requests.get(url)
+
+    soup = BeautifulSoup(r.text, 'html.parser')
+
+    # find the table with the CVEs
+    table = soup.find('table', attrs={'class': 'risk-matrix'})
+
+    # find all the rows in the table
+    rows = table.find_all('tr')
+
+    # fetch CVE data from first td in each row
+    for row in rows:
+
+        # find the versions in the first row
+        header = row.find('th')
+        versions = []
+        if header is not None:
+            component = header.find_next_sibling('th')
+            if component.text == 'Component':
+                score = component.find_next_sibling('th')
+                while (score.find_next_sibling('th') is not None):
+                    versions.append(score.find_next_sibling('th').text)
+                    score = score.find_next_sibling('th')
+
+        cve = row.find('td')
+        if cve is not None:
+            
+            id = cve.text
+            link = cve.find('a')['href']
+            componentsTD = cve.find_next_sibling('td')
+            component = componentsTD.text.replace('\n', '')
+            scoreTD = componentsTD.find_next_sibling('td')
+            score = scoreTD.text
+
+            versionCheck = scoreTD
+            affected_versions = []
+            for version in versions:
+                versionCheck = versionCheck.find_next_sibling('td')
+                if versionCheck.text == '•':
+                    affected_versions.append(int(version))
+
+            # make an API call to get the additional data for the CVE
+            '''
+            '''
+            affects = BomTarget(
+                ref=component
+            )
+            for v in affected_versions:
+                affects.versions.add(v)
+            vuln = Vulnerability(
+                id=id,
+                source=VulnerabilitySource(name="National Vulnerability Database", url=link),
+                #todo: dummy date
+                published=datetime.fromisoformat(date),
+                updated=datetime.fromisoformat(date),
+                description="",
+                recommendation=""
+            )
+            vuln.affects.add(affects)
+            vulnerabilities.append(vuln)
+            print(vuln)
+    return vulnerabilities
```

Comment:
Do you want to build a BOM here, or just return a simple data structure that captures the info scraped from the website?

If BOMs provide all you need then fine, but simplify where you can at this stage IMHO.